### ISSUE TITLE
Openbox right click menu not working with glava

### ISSUE THREAD

**Issue by InvisibleRasta** [wacossusca34/glava#80]
hello guys have you figured out how to get the openbox right click menu to work with glava? i tried all the setxwintype options and none of them worked

**Comment by wacossusca34** [comment]
You probably want `#request setclickthrough true`. This is auto-enabled for `#request setxwintype "desktop"`, but if you're using a different window type, then you will need to set that option yourself.

If these options do not allow clicking through the window, then this is a bug.

**Comment by InvisibleRasta** [comment]
yeah i tried with desktop and it didnt let me click with openbox on Archlinux

**Comment by InvisibleRasta** [comment]
openbox 3.6.1-4

**Comment by InvisibleRasta** [comment]
have you been able to give it a look?

**Comment by wacossusca34** [comment]
@InvisibleRasta I am busy this week taking first aid courses.

**Comment by wacossusca34** [comment]
@InvisibleRasta this was a pain in the ass to fix, since there's not a whole lot of information on using XShape and how WMs are supposed to respond. Turns out _two things_ were breaking clickthrough:

* The way I was calling `apply_clickthrough` after mapping wasn't reliable on all window managers (after `XMapWindow`). Calling this on `MapNotify` fixes this for Openbox.

* For some absurd reason, desktop windows on Openbox have an invisible (and pointless?) parent window. I fixed this by walking through the parent windows in `apply_clickthrough`.

For unmanaged windows, clickthrough worked already worked fine, but this fix should clean up any remaining XShape issues.
